tests/reader_test.py

```diff
@@ -29,6 +29,8 @@ def test_data_file(input_cif):
         reader = PdbxReader(input_file)
         data_list = []
         reader.read(data_list)
+        for item in data_list:
+            print(item.get_object("atom_site").print_it())
```

Comment:
`print_it()` doesn't return a string, so won't this print `None`? Is this intentional?

Reply:
@speleo3 it was intentional since I run `python -m pytest -s tests` to get the following output from `print_it()`:

```
tests/reader_test.py --------------------------------------------
 Category: atom_site attribute list length: 25
 Category: atom_site attribute: group_PDB
 Category: atom_site attribute: id
 Category: atom_site attribute: type_symbol
 Category: atom_site attribute: label_atom_id
 Category: atom_site attribute: label_alt_id
 Category: atom_site attribute: label_comp_id
 Category: atom_site attribute: label_asym_id
 Category: atom_site attribute: label_entity_id
 Category: atom_site attribute: label_seq_id
 Category: atom_site attribute: pdbx_PDB_ins_code
 Category: atom_site attribute: Cartn_x
 Category: atom_site attribute: Cartn_y
 Category: atom_site attribute: Cartn_z
 Category: atom_site attribute: occupancy
 Category: atom_site attribute: B_iso_or_equiv
 Category: atom_site attribute: Cartn_x_esd
 Category: atom_site attribute: Cartn_y_esd
 Category: atom_site attribute: Cartn_z_esd
 Category: atom_site attribute: occupancy_esd
 Category: atom_site attribute: B_iso_or_equiv_esd
 Category: atom_site attribute: auth_seq_id
 Category: atom_site attribute: auth_comp_id
 Category: atom_site attribute: auth_asym_id
 Category: atom_site attribute: auth_atom_id
 Category: atom_site attribute: pdbx_PDB_model_num
 Row value list length: 2840
                      group_PDB: ATOM ...
                             id: 1 ...
                    type_symbol: N ...
                  label_atom_id: N ...
                   label_alt_id:  ...
                  label_comp_id: ASP ...
                  label_asym_id: A ...
                label_entity_id: 1 ...
                   label_seq_id: 1 ...
              pdbx_PDB_ins_code: None ...
                        Cartn_x: 23.482 ...
                        Cartn_y: -0.621 ...
                        Cartn_z: -1.419 ...
                      occupancy: 1.00 ...
                 B_iso_or_equiv: 35.27 ...
                    Cartn_x_esd: None ...
                    Cartn_y_esd: None ...
                    Cartn_z_esd: None ...
                  occupancy_esd: None ...
             B_iso_or_equiv_esd: None ...
                    auth_seq_id: 1 ...
                   auth_comp_id: ASP ...
                   auth_asym_id: A ...
                   auth_atom_id: N ...
             pdbx_PDB_model_num: 1 ...
                      group_PDB: ATOM ...
                             id: 2 ...
                    type_symbol: C ...
                  label_atom_id: CA ...
                   label_alt_id:  ...
                  label_comp_id: ASP ...
                  label_asym_id: A ...
                label_entity_id: 1 ...
                   label_seq_id: 1 ...
              pdbx_PDB_ins_code: None ...
                        Cartn_x: 24.897 ...
                        Cartn_y: -0.728 ...
                        Cartn_z: -1.885 ...
                      occupancy: JUNK ...
                 B_iso_or_equiv: 32.46 ...
                    Cartn_x_esd: None ...
                    Cartn_y_esd: None ...
                    Cartn_z_esd: None ...
                  occupancy_esd: None ...
             B_iso_or_equiv_esd: None ...
                    auth_seq_id: 1 ...
                   auth_comp_id: ASP ...
                   auth_asym_id: A ...
                   auth_atom_id: CA ...
             pdbx_PDB_model_num: 1 ...
None
.--------------------------------------------
 Category: atom_site attribute list length: 25
 Category: atom_site attribute: group_PDB
 Category: atom_site attribute: id
 Category: atom_site attribute: type_symbol
 Category: atom_site attribute: label_atom_id
 Category: atom_site attribute: label_alt_id
 Category: atom_site attribute: label_comp_id
 Category: atom_site attribute: label_asym_id
 Category: atom_site attribute: label_entity_id
 Category: atom_site attribute: label_seq_id
 Category: atom_site attribute: pdbx_PDB_ins_code
 Category: atom_site attribute: Cartn_x
 Category: atom_site attribute: Cartn_y
 Category: atom_site attribute: Cartn_z
 Category: atom_site attribute: occupancy
 Category: atom_site attribute: B_iso_or_equiv
 Category: atom_site attribute: Cartn_x_esd
 Category: atom_site attribute: Cartn_y_esd
 Category: atom_site attribute: Cartn_z_esd
 Category: atom_site attribute: occupancy_esd
 Category: atom_site attribute: B_iso_or_equiv_esd
 Category: atom_site attribute: auth_seq_id
 Category: atom_site attribute: auth_comp_id
 Category: atom_site attribute: auth_asym_id
 Category: atom_site attribute: auth_atom_id
 Category: atom_site attribute: pdbx_PDB_model_num
 Row value list length: 64281
                      group_PDB: ATOM ...
                             id: 1 ...
                    type_symbol: O ...
                  label_atom_id: O5' ...
                   label_alt_id:  ...
                  label_comp_id: U ...
                  label_asym_id: A ...
                label_entity_id: 1 ...
                   label_seq_id: 9 ...
              pdbx_PDB_ins_code: None ...
                        Cartn_x: 17.684 ...
                        Cartn_y: 150.891 ...
                        Cartn_z: 105.893 ...
                      occupancy: 1.00 ...
                 B_iso_or_equiv: 57.93 ...
                    Cartn_x_esd: None ...
                    Cartn_y_esd: None ...
                    Cartn_z_esd: None ...
                  occupancy_esd: None ...
             B_iso_or_equiv_esd: None ...
                    auth_seq_id: 10 ...
                   auth_comp_id: U ...
                   auth_asym_id: 0 ...
                   auth_atom_id: O5' ...
             pdbx_PDB_model_num: 1 ...
                      group_PDB: ATOM ...
                             id: 2 ...
                    type_symbol: C ...
                  label_atom_id: C5' ...
                   label_alt_id:  ...
                  label_comp_id: U ...
                  label_asym_id: A ...
                label_entity_id: 1 ...
                   label_seq_id: 9 ...
              pdbx_PDB_ins_code: None ...
                        Cartn_x: 17.611 ...
                        Cartn_y: 149.829 ...
                        Cartn_z: 104.909 ...
                      occupancy: 1.00 ...
                 B_iso_or_equiv: 59.38 ...
                    Cartn_x_esd: None ...
                    Cartn_y_esd: None ...
                    Cartn_z_esd: None ...
                  occupancy_esd: None ...
             B_iso_or_equiv_esd: None ...
                    auth_seq_id: 10 ...
                   auth_comp_id: U ...
                   auth_asym_id: 0 ...
                   auth_atom_id: C5' ...
             pdbx_PDB_model_num: 1 ...
None
...........s.
tests/readwrite_test.py --------------------------------------------
 Category: category attribute list length: 10
 Category: category attribute: aOne
 Category: category attribute: aTwo
 Category: category attribute: aThree
 Category: category attribute: aFour
 Category: category attribute: aFive
 Category: category attribute: aSix
 Category: category attribute: aSeven
 Category: category attribute: aEight
 Category: category attribute: aNine
 Category: category attribute: aTen
 Row value list length: 10
                           aOne: 1 ...
                           aTwo: 2 ...
                         aThree: 3 ...
                          aFour: 4 ...
                          aFive: 5 ...
                           aSix: 6 ...
                         aSeven: 7 ...
                         aEight: 8 ...
                          aNine: 9 ...
                           aTen: 10 ...
                           aOne: 1 ...
                           aTwo: 2 ...
                         aThree: 3 ...
                          aFour: 4 ...
                          aFive: 5 ...
                           aSix: 6 ...
                         aSeven: 7 ...
                         aEight: 8 ...
                          aNine: 9 ...
                           aTen: 10 ...
+ data container:                        myblock contains    1 categories
--------------------------------------------
Data category: category
--------------------------------------------
 Category: category attribute list length: 10
 Category: category attribute: aOne
 Category: category attribute: aTwo
 Category: category attribute: aThree
 Category: category attribute: aFour
 Category: category attribute: aFive
 Category: category attribute: aSix
 Category: category attribute: aSeven
 Category: category attribute: aEight
 Category: category attribute: aNine
 Category: category attribute: aTen
 Row value list length: 10
                           aOne: 1 ...
                           aTwo: 2 ...
                         aThree: 3 ...
                          aFour: 4 ...
                          aFive: 5 ...
                           aSix: 6 ...
                         aSeven: 7 ...
                         aEight: 8 ...
                          aNine: 9 ...
                           aTen: 10 ...
                           aOne: 1 ...
                           aTwo: 2 ...
                         aThree: 3 ...
                          aFour: 4 ...
                          aFive: 5 ...
                           aSix: 6 ...
                         aSeven: 7 ...
                         aEight: 8 ...
                          aNine: 9 ...
                           aTen: 10 ...
.+ data container:                        myblock contains    1 categories
--------------------------------------------
Data category: pdbx_seqtool_mapping_ref
--------------------------------------------
 Category: pdbx_seqtool_mapping_ref attribute list length: 7
 Category: pdbx_seqtool_mapping_ref attribute: ordinal
 Category: pdbx_seqtool_mapping_ref attribute: entity_id
 Category: pdbx_seqtool_mapping_ref attribute: auth_mon_id
 Category: pdbx_seqtool_mapping_ref attribute: auth_mon_num
 Category: pdbx_seqtool_mapping_ref attribute: pdb_chain_id
 Category: pdbx_seqtool_mapping_ref attribute: ref_mon_id
 Category: pdbx_seqtool_mapping_ref attribute: ref_mon_num
 Row value list length: 4
                        ordinal: 9 ...
                      entity_id: 2 ...
                    auth_mon_id: 3 ...
                   auth_mon_num: 4 ...
                   pdb_chain_id: 5 ...
                     ref_mon_id: 6 ...
                    ref_mon_num: 7 ...
                        ordinal: 10 ...
                      entity_id: 2 ...
                    auth_mon_id: 3 ...
                   auth_mon_num: 4 ...
                   pdb_chain_id: 5 ...
                     ref_mon_id: 6 ...
                    ref_mon_num: 7 ...
--------------------------------------------
 Category: pdbx_seqtool_mapping_ref attribute list length: 7
 Category: pdbx_seqtool_mapping_ref attribute: ordinal
 Category: pdbx_seqtool_mapping_ref attribute: entity_id
 Category: pdbx_seqtool_mapping_ref attribute: auth_mon_id
 Category: pdbx_seqtool_mapping_ref attribute: auth_mon_num
 Category: pdbx_seqtool_mapping_ref attribute: pdb_chain_id
 Category: pdbx_seqtool_mapping_ref attribute: ref_mon_id
 Category: pdbx_seqtool_mapping_ref attribute: ref_mon_num
 Row value list length: 4
                        ordinal: 9 ...
                      entity_id: 2 ...
                    auth_mon_id: 3 ...
                   auth_mon_num: 4 ...
                   pdb_chain_id: 5 ...
                     ref_mon_id: 6 ...
                    ref_mon_num: 7 ...
                        ordinal: 10 ...
                      entity_id: 2 ...
                    auth_mon_id: 3 ...
                   auth_mon_num: 4 ...
                   pdb_chain_id: 5 ...
                     ref_mon_id: 6 ...
                    ref_mon_num: 7 ...
```